src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
+            }
+            sizeX = -1.0f;
+            sizeY -= step;
+        }
+
+        GLuint pixelSizeQuadVAO = 0;
+        GLuint pixelSizeQuadVBO = 0;
 
-    const std::vector<Deferredcaster*>& deferredcasters =
-        OsEng.renderEngine().deferredcasterManager().deferredcasters();
-    int nextId = 0;
-    for (auto& caster : deferredcasters) {
-        DeferredcastData data;
-        data.id = nextId++;
-        data.namespaceName = "HELPER";
+        glGenVertexArrays(1, &pixelSizeQuadVAO);
+        glBindVertexArray(pixelSizeQuadVAO);
 
-        std::string vsPath = caster->deferredcastVSPath();
-        std::string fsPath = caster->deferredcastFSPath();
-        std::string deferredShaderPath = caster->deferredcastPath();
+        glGenBuffers(1, &pixelSizeQuadVBO);
+        glBindBuffer(GL_ARRAY_BUFFER, pixelSizeQuadVBO);
 
-        ghoul::Dictionary dict;
-        dict.setValue("rendererData", _rendererData);
-        //dict.setValue("fragmentPath", fsPath);
-        dict.setValue("id", data.id);
-        std::string helperPath = caster->helperPath();
-        ghoul::Dictionary helpersDict;
-        if (helperPath != "") {
-            helpersDict.setValue("0", helperPath);
-        }
-        dict.setValue("helperPaths", helpersDict);
-        //dict.setValue("deferredcastPath", caster->getDeferredcastPath());
+        glBufferData(
+            GL_ARRAY_BUFFER,
+            sizeof(GLfloat) * GRIDSIZE * GRIDSIZE * NVERTEX,
+            vertexData,
+            GL_STATIC_DRAW
+        );
+
+        // Position
+        glVertexAttribPointer(
+            0,
+            4,
+            GL_FLOAT,
+            GL_FALSE,
+            0,
+            nullptr
+        );
+        glEnableVertexAttribArray(0);
+
+        // Saves current state
+        GLint defaultFbo;
+        glGetIntegerv(GL_FRAMEBUFFER_BINDING, &defaultFbo);
+
+        glm::ivec4 viewport = OsEng.windowWrapper().viewportPixelCoordinates();
+
+        // Main framebuffer
+        GLuint pixelSizeTexture = 0;
+        GLuint pixelSizeFramebuffer = 0;
+
+        glGenTextures(1, &pixelSizeTexture);
+        glBindTexture(GL_TEXTURE_2D_MULTISAMPLE, pixelSizeTexture);
+
+        const GLsizei ONEPIXEL = 1;
+        glTexImage2DMultisample(
+            GL_TEXTURE_2D_MULTISAMPLE,
+            _nAaSamples,
+            GL_RGBA32F,
+            ONEPIXEL,
+            ONEPIXEL,
+            true
+        );
 
-        _deferredcastData[caster] = data;
+        glViewport(0, 0, ONEPIXEL, ONEPIXEL);
 
+        glGenFramebuffers(1, &pixelSizeFramebuffer);
+        glBindFramebuffer(GL_FRAMEBUFFER, pixelSizeFramebuffer);
+        glFramebufferTexture2D(
+            GL_FRAMEBUFFER,
+            GL_COLOR_ATTACHMENT0,
+            GL_TEXTURE_2D_MULTISAMPLE,
+            pixelSizeTexture,
+            0
+        );
+
+        GLenum textureBuffers[1] = { GL_COLOR_ATTACHMENT0 };
+        glDrawBuffers(1, textureBuffers);
+
+        glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
```

Comment:
`0.f`

Reply:
:-)

Reply:
Picky? :-D

Reply:
Done.

Reply:
Hahah..  yeah ;)  Code is a work of art and my OCD kicks in sometimes ;)